src/blocks/remote-data-no-result/edit.tsx

```diff
@@ -0,0 +1,52 @@
+/**
+ * WordPress dependencies
+ */
+import { useBlockProps, useInnerBlocksProps } from '@wordpress/block-editor';
+import { BlockEditProps, Template } from '@wordpress/blocks';
+
+import { useRemoteDataContext } from '@/blocks/remote-data-container/hooks/useRemoteDataContext';
+import { __ } from '@/utils/i18n';
+
+import './editor.scss';
+
+const NO_RESULTS_TEMPLATE: Template[] = [
+	[
+		'core/paragraph',
+		{
+			content: __( 'No results found.' ),
+		},
+	],
+];
+
+const NO_RESULTS_PLACEHOLDER_TEMPLATE: Template[] = [
+	[
+		'core/paragraph',
+		{
+			placeholder: __(
+				'This block only works when placed inside a remote data block when there are no results. This block will be ignored as currently configured.'
+			),
+		},
+	],
+];
+
+export function Edit( props: BlockEditProps< RemoteDataNoResultBlockAttributes > ): JSX.Element {
+	const { context } = props;
+	const { remoteData } = useRemoteDataContext( context );
+	const blockProps = useBlockProps();
+
+	// This is mirroring the query-no-results block from Gutenberg.
+	// https://github.com/WordPress/gutenberg/blob/trunk/packages/block-library/src/query-no-results/edit.js
+	let templateToUse = NO_RESULTS_PLACEHOLDER_TEMPLATE;
+
+	if ( remoteData?.results?.length === 0 ) {
+		templateToUse = NO_RESULTS_TEMPLATE;
+	} else if ( remoteData?.results ) {
+		templateToUse = [];
+	}
+
```

Comment:
I don't think this logic is useful. Similar to the core block, we always want the editor to see and edit the content that _would_ display if there are no results, regardless of whether there are _currently_ any results.

```suggestion
```

Reply:
Fair enough, it is a state that we want them to configure so it should always be visible.